addon/errors.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Checks if the given response is a FetchError
+ */
+export function isFetchError(response: Response): boolean {
+  return response.ok === false
```

Comment:
I think things are different in `fetch`.

`response.ok` is true for status 2xx, which will be false for 304 https://github.com/github/fetch/issues/241. I think we should remove this function and just use `isSuccess` you added at bottom.

Another thing to notice is that all the response check are successful in regards to network/abort and checking status from server. We also need to deal with `.catch` scenarios, where `response.type === 'error'` and `aborted flag` ([spec](https://fetch.spec.whatwg.org/#concept-response-type)).